source/Missions/Services/Agents/Handlers/AgentDamageHandler.cs

```diff
@@ -0,0 +1,112 @@
+﻿using Common;
+using Common.Messaging;
+using Common.Network;
+using GameInterface.Serialization;
+using LiteNetLib;
+using Missions.Services.Agents.Messages;
+using Missions.Services.Agents.Packets;
+using Missions.Services.Network;
+using System;
+using TaleWorlds.MountAndBlade;
+
+namespace Missions.Services.Agents.Handlers
+{
+    /// <summary>
+    /// Handler for agent damage in a battle
+    /// </summary>
+    public interface IAgentDamageHandler : IHandler
```

Comment:
Add inheritance to IDisposable
Do the same for the other handlers in the PR as well
```suggestion
    public interface IAgentDamageHandler : IHandler, IDisposable
```